src/sagas/options.js

```diff
@@ -0,0 +1,228 @@
+    return undefined
+  }
+  const deafultOptionsTemplate = fromJS(defaultOptions)
+  const defaultOptionsForAllPaths = paths.reduce((acc, path) => acc.setIn(path, deafultOptionsTemplate), fromJS({}))
+
+  return currentOptions.mergeDeepWith((prev, next, key) => {
+    if (next === undefined) {
+      console.error(`Unknown property ${key} has no default value`)
+    }
+    if (prev !== undefined) {
+      // use existing property if present
+      return prev
+    }
+    // use default
+    return next
+  }, defaultOptionsForAllPaths)
+}
+
+function* saveRemoteOptions (newOptions: Object, paths: Array<Array<string>>, defaultOptions?: Object): any {
+  const updatedProps = Object.entries(newOptions).filter(([ name, value ]) => value !== undefined)
+  if (!updatedProps.length) {
+    return { changes: [] }
+  }
+  const updatedPropNames = updatedProps.map(([key, value]) => key)
+
+  const currentOptions = yield select((state) => state.options)
+  const mergedWithDefaults = mergeWithDefaults(currentOptions, paths, defaultOptions) || currentOptions
+
+  const newOptionsTemplate = fromJS(Object.fromEntries(updatedProps))
+  const newOptionsForAllPaths = paths.reduce((acc, path) => acc.setIn(path, newOptionsTemplate), fromJS({}))
+  const mergedWithDefaultsAndNew = mergedWithDefaults.mergeDeepWith((prev, next, key) => {
+    if (next === undefined) {
+      // no new value for this property so use existing
+      return prev
+    }
+    // use new value
+    return next
+  }, newOptionsForAllPaths)
+
+  if (mergedWithDefaultsAndNew.equals(currentOptions)) {
+    return { changes: updatedPropNames, sameAsCurrent: true }
+  }
+
+  const receivedOptions = yield select((state) => state.config.getIn(['user', 'receivedOptions']))
+  const mergedWithReceived = Api.userOptionsToApi(mergedWithDefaultsAndNew.toJS(), receivedOptions.toJS())
+  const userId = yield select((state) => state.config.getIn(['user', 'id']))
+
+  const result = yield * callExternalAction(
+    'saveUserOptionsOnBackend',
+    Api.saveUserOptionsOnBackend,
+    saveUserOptionsOnBackend({ options: mergedWithReceived, userId }),
+    true)
+  return { ...result, changes: updatedPropNames }
+}
+
+function withLoadingUserOptions (delegateGenerator: (any) => Generator<any, any, any>): any {
+  return function* (action: any): any {
+    yield put(loadingUserOptionsInProgress())
+    try {
+      yield call(delegateGenerator, action)
+    } finally {
+      yield put(loadingUserOptionsFinished())
+    }
+  }
+}
+
+export function* saveGlobalOptions ({ payload: { sshKey, showNotifications, notificationSnoozeDuration, language, updateRate }, meta: { correlationId } }: SaveGlobalOptionsActionType): Generator<any, any, any> {
+  const [
+    { error: sshError, changes: sshChanges = [], sameAsCurrent: keySameAsCurrent, ...sshData },
+    { error: remoteOptionsError, changes: remoteOptionChanges = [], sameAsCurrent: remoteOptionsSameAsCurrent, ...userData },
+  ] = yield all([
```

Comment:
Still need to wrap the sagas in `call` if you want them to run in parallel.

Leave it as is if they need to run sequentially or it doesn't matter.  Although, I'd prefer to see them explicitly called in sequence with the simple form:

```js
const { ... } = yield saveSSHKey(...)
const { ... } = yield saveRemoteOptions(...)
```

Reply:
the intention was to run them in parallel and it seems it works but let's switch to `call()` syntax just to be on the safe side